### ISSUE TITLE
Automatically dismiss the current game when creating a follow-up game

### ISSUE THREAD

**Issue by irilyth** [buttonmen-dev/buttonmen#1657]
When I use the buttons to create a follow-up game (the ones after "Challenge X to another game, preserving chat" on the completed-game page), I invariably want to dismiss the original game immediately afterwards. (If I ever want to look at it again, I can always find it in the follow-up game's chat log.) It'd be cool if this happened automatically; perhaps as a user preference, if we're not confident that everyone wants this.

**Comment by irilyth** [comment]
This is my BFOTM for September.

**Comment by blackshadowshade** [comment]
What do you think about addressing this by having a broader user preference: "Auto-dismiss all completed games"? Or does this go too far?

**Comment by irilyth** [comment]
Yeah, that's not really what I want -- in particular, sometimes my opponents don't create follow-up games, and I want to, so if those get auto-dismissed, I'll never see them.

I think it's fine if this isn't a user preference; I'd be surprised if anyone wanted to create a follow-up game and *not* dismiss the original. It's easy to find the original from the follow-up. We could ask, and/or make a preference later if anyone turns out to care.

**Comment by blackshadowshade** [comment]
I'd like to hear what people on the live site say about this. Feel like starting a thread on the forums? Once I hear what people are saying, that'll help me with whether to add a user preference or not.

**Comment by blackshadowshade** [comment]
If we go down the route of not having a user preference, I'll add some wording somewhere that says that creating the follow-up game automatically dismisses the old one.

**Comment by irilyth** [comment]
Sure, will do.

**Comment by irilyth** [comment]
One other thought about the interface to this: The Create Game screen already has information about the fact that it's creating a follow-up game (the "Copy chat from: Game X" field); one way to do this might be to add a "Dismiss game X" checkbox right under that, which could be checked by default, and you could uncheck it if you didn't want that for whatever reason. That would also have the advantage of only dismissing the game at create time, rather than when you click on the follow-up link, which could be annoying if you clicked the link by mistake.

**Comment by blackshadowshade** [comment]
So, in the solution that I've submitted here, every shortcut link that creates a game with preserved chat automatically dismisses the completed game, but only when the person actually creates the game. This gets around the issue of accidentally hitting the link.

**Comment by blackshadowshade** [comment]
A number of solution attempts are found in #1742, but this looks like it could well be somewhat involved with the way that the API works with messages at the moment.
